Tools/ViewsEstimators.py

```diff
@@ -6,118 +6,333 @@
+
+                    'LGBMRegressor': LGBMRegressor(n_estimators=250, random_state=random_state),
+                    'LGBMClassifier': LGBMClassifier(n_estimators=250, random_state=random_state),
+
+                    'RFRegressor': RandomForestRegressor(n_estimators=250, random_state=random_state),
+                    'RFClassifier': RandomForestClassifier(n_estimators=250, random_state=random_state),
+
+                    'XGBRFRegressor': XGBRFRegressor(n_estimators=100, learning_rate=0.05, random_state=random_state),
+                    'XGBRFClassifier': XGBRFClassifier(n_estimators=100, learning_rate=0.05, random_state=random_state),
+
+                    'GBMRegressor': GradientBoostingRegressor(n_estimators=200, random_state=random_state),
+                    'GBMClassifier': GradientBoostingClassifier(n_estimators=200, random_state=random_state),
+
+                    'XGBRegressor': XGBRegressor(n_estimators=100, learning_rate=0.05, random_state=random_state),
+                    'XGBClassifier': XGBClassifier(n_estimators=100, learning_rate=0.05, random_state=random_state),
+                    
+                    'HGBRegressor': HistGradientBoostingRegressor(max_iter=200, random_state=random_state),
+                    'HGBClassifier': HistGradientBoostingClassifier(max_iter=200, random_state=random_state),
                 }
+        estimator = estimators.get(estimator_name)
 
-        return funcs[func_name]
+        return estimator
 
+    # Define the fit method for the class to train the model
     def fit(self,
             X: Union[np.ndarray, pd.DataFrame],
             y: Union[np.ndarray, pd.Series]):
-        X, y = check_X_y(X, y, dtype=None,
+
+        # Use sklearn fucntionality to check X and y for consistent length and enforce X to be 2D and y 1D. 
+        # By default, X is checked to be non-empty and containing only finite values.
+        # Standard input checks are also applied to y, such as checking that y
+        # does not have np.nan or np.inf targets.
+        X, y = check_X_y(X, 
+                         y, 
+                         dtype=None,
                          accept_sparse=False,
                          accept_large_sparse=False,
-                         force_all_finite='allow-nan')
+                         force_all_finite=True) #'allow-nan'
+        
+        # Set the number of features seen during fit
+        self.n_features_in_ = X.shape[1]
 
         if X.shape[1] < 2:
             raise ValueError('Cannot fit model when n_features = 1')
 
-        self.clf_ = self._resolve_estimator(self.clf_name)
+        # Instantiate the classifier
+        self.clf_ = self._resolve_estimator(self.clf_name, random_state=42)
         if self.clf_params:
             self.clf_.set_params(**self.clf_params)
-        self.clf_.fit(X, y > 0)
-        self.clf_fi = self.clf_.feature_importances_
+        
+        # Check if there are more than one unique values in y 
```

Comment:
I don't get this.  If there is only one y then something must have gone wrong way back in the data part.  I think it is prudent with these kinds of checks, but this is a bit late to do it no? And why try to fit it if the data is wrong? I.e. with the DummyClassifier?

Reply:
Good question! I made some changes here already, but I will explain why I was fumbling with checking whether a training set has 1 sample only:
- I called `check_estimator(hr)` function inside the test function to do automatic sanity checks on the estimator. `check_estimator()` is an `sklearn` native check and among other tests it runs test called `check_fit2d_1sample` to make sure that the estimator can handle an edge case where only one sample is passed to the fitting stage. So, to pass this automatic sklearn test within my test function, I needed to do something hacky in the fit method :) 
- I decided to simply delete `check_estimator(hr)`, so I do not need to have a hacky work around to pass it. We can add explicit tests for the estimator into the testing function if needed.